dbt/adapters/redshift/relation_configs/dist.py

```diff
@@ -0,0 +1,164 @@
+                ),
+                validation_error=DbtRuntimeError(
+                    "A `RedshiftDistConfig` that specifies a `distkey` must be of `diststyle` `key`."
+                ),
+            ),
+        }
+
+    @classmethod
+    def from_dict(cls, config_dict) -> "RedshiftDistConfig":
+        kwargs_dict = {
+            "diststyle": config_dict.get("diststyle"),
+            "distkey": config_dict.get("distkey"),
+        }
+        dist: "RedshiftDistConfig" = super().from_dict(kwargs_dict)  # type: ignore
+        return dist
+
+    @classmethod
+    def parse_model_node(cls, model_node: ModelNode) -> dict:
+        """
+        Translate ModelNode objects from the user-provided config into a standard dictionary.
+
+        Args:
+            model_node: the description of the distkey and diststyle from the user in this format:
+
+                {
+                    "dist": any("auto", "even", "all") or "<column_name>"
+                }
+
+        Returns: a standard dictionary describing this `RedshiftDistConfig` instance
+        """
+        dist = model_node.config.extra.get("dist", "")
+
+        if dist == "":
+            return {}
+
+        elif dist.lower() in (
+            RedshiftDistStyle.auto,
+            RedshiftDistStyle.even,
+            RedshiftDistStyle.all,
+        ):
+            return {"diststyle": dist.lower()}
+
+        else:
+            return {"diststyle": RedshiftDistStyle.key, "distkey": dist}
+
+    @classmethod
+    def parse_relation_results(cls, relation_results_entry: agate.Row) -> dict:
+        """
+        Translate agate objects from the database into a standard dictionary.
+
+        Args:
+            relation_results_entry: the description of the distkey and diststyle from the database in this format:
+
+                agate.Row({
+                    "diststyle": "<diststyle/distkey>",  # e.g. EVEN | KEY(column1) | AUTO(ALL) | AUTO(KEY(id))
+                })
+
+        Returns: a standard dictionary describing this `RedshiftDistConfig` instance
+        """
+        dist: str = relation_results_entry.get("diststyle", "")
+
+        try:
+            diststyle = dist.split("(")[0]  # covers `AUTO`, `ALL`, `KEY`, <null>, <unexpected>
+        except ValueError:
+            diststyle = dist  # covers `EVEN`, <unexpected>
+
+        if diststyle == "":
+            return {}
+
+        elif diststyle.lower() in [
+            RedshiftDistStyle.auto,
+            RedshiftDistStyle.all,
+            RedshiftDistStyle.even,
+        ]:
+            return {"diststyle": diststyle.lower()}
+
+        elif diststyle == RedshiftDistStyle.key:
+            open_paren = len("KEY(")
+            close_paren = -len(")")
+            distkey = dist[open_paren:close_paren]  # e.g. KEY(column1)
+
+            return {"diststyle": diststyle.lower(), "distkey": distkey}
+
+        else:
+            raise DbtRuntimeError(f"Unexpected dist format: {dist}")
```

Comment:
Unlikely edge case but do we want to treat an unexpected dist style from redshift as a runtime error? The scenario I'm imagining here being that redshift adds a new diststyle or changes the returned result and dbt jobs will start to fail. 
If we want to treat this as an exception maybe we add a little more detail: `Recieved an unexpected dist format from database: {dist}`?

Reply:
Honestly, I think it's more likely that we just won't cover all possible cases that Redshift can produce right out of the gate, rather than there being a truly new diststyle. That being said, I agree the message is vague. I'll update it.